src/backend/common/helpers/matchstats_helper.py

```diff
@@ -8,29 +8,146 @@
 
 # x is OPR and should be n x 1
 
-from collections import defaultdict
-from typing import Any, Dict, List, Optional, Tuple
+from collections import OrderedDict
+from typing import Callable, Dict, List, Tuple
 
 import numpy as np
-from google.cloud import ndb
 from pyre_extensions import none_throws
 
-from backend.common.consts.alliance_color import ALLIANCE_COLORS, AllianceColor
+from backend.common.consts.alliance_color import (
+    ALLIANCE_COLORS,
+    AllianceColor,
+    OPPONENT,
+)
 from backend.common.consts.comp_level import CompLevel
-from backend.common.consts.event_type import EventType
-from backend.common.memcache import MemcacheClient
-from backend.common.models.keys import TeamId, Year
+from backend.common.models.event_matchstats import (
+    Component,
+    EventMatchstats,
+    StatType,
+    TeamStatMap,
+)
+from backend.common.models.keys import TeamId
 from backend.common.models.match import Match
-from backend.common.models.stats import EventMatchStats, StatType
-from backend.common.queries import event_query
-
 
+StatAccessor = Callable[[Match, AllianceColor], float]
 TTeamIdMap = Dict[TeamId, int]
 
 
-class MatchstatsHelper(object):
+class MatchstatsHelper:
+    OPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[color]["score"]
+    DPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[OPPONENT[color]][
+        "score"
+    ]
+    CCWM_ACCESSOR: StatAccessor = lambda match, color: (
+        match.alliances[color]["score"] - match.alliances[OPPONENT[color]]["score"]
+    )
+    DEFAULT_COMPONENT_ACCESSOR: Callable[[Component], StatAccessor] = lambda comp: (
+        lambda match, color: match.score_breakdown[color].get(comp, 0)
+    )
+
+    EVERGREEN: Dict[StatType, StatAccessor] = {
+        StatType.OPR: OPR_ACCESSOR,
+        StatType.DPR: DPR_ACCESSOR,
+        StatType.CCWM: CCWM_ACCESSOR,
+    }
+
+    COMPONENTS = {
+        2020: {
+            "Total Power Cell Points": lambda match, color: (
+                match.score_breakdown[color].get("autoCellPoints")
+                + match.score_breakdown[color].get("teleopCellPoints")
+            ),
+        },
+        2019: {
+            "Cargo + Panel Points": lambda match, color: (
+                match.score_breakdown[color].get("cargoPoints")
+                + match.score_breakdown[color].get("hatchPanelPoints")
+            )
+        },
+    }
```

Comment:
We should add a `Pre-Competition Season` section to the [New Season](https://github.com/the-blue-alliance/the-blue-alliance/wiki/New-Season) page of the runbook and make a note to update the component OPR, along with some details around how we should pick which thing(s) to show for component OPRs. Those pages are markdown files located in the `docs/` folder https://github.com/the-blue-alliance/the-blue-alliance/blob/py3/docs/Maintainers/New-Season.md

This section needs to be added anyways for things like match breakdowns, rankings, etc. We just haven't gotten that stuff moved to py3 yet, so that section doesn't exist yet.